recipes/metis/all/conanfile.py

```diff
@@ -14,88 +16,78 @@ class METISConan(ConanFile):
                   " partitioning finite element meshes, and producing" \
                   " fill reducing orderings for sparse matrices"
     topics = ("karypislab", "graph", "partitioning-algorithms")
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "with_64bit_types": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "with_64bit_types": True,
     }
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
 
     @property
     def _is_mingw(self):
         return self.settings.os == "Windows" and self.settings.compiler == "gcc"
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("gklib/5.1.1")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _patch_sources(self):
-        apply_conandata_patches(self)
-
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
-            self._cmake.definitions["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
-            self._cmake.definitions["SHARED"] = self.options.shared
-            self._cmake.definitions["METIS_INSTALL"] = True
-            self._cmake.definitions["ASSERT"] = self.settings.build_type == "Debug"
-            self._cmake.definitions["ASSERT2"] = self.settings.build_type == "Debug"
-            self._cmake.definitions["METIS_IDX64"] = True
-            self._cmake.definitions["METIS_REAL64"] = True
-            self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
```

Comment:
No need to set this, already done by the CMakeToolchain integration :)

Reply:
It's required when cmake_minimum_required is less than 3.0, if you want to set install_name of shared lib to `@rpath` on Apple (relocatable shared lib). CMakeToolchain does not manage this policy.

Reply:
It would be enough, didn't see it. What is the rational behind this bump of cmake min version? It's not explained.